pkg/chartverifier/checks/registry.go

```diff
@@ -42,21 +42,28 @@ func (r *Result) SetResult(outcome bool, reason string) Result {
 func (r *Result) AddResult(outcome bool, reason string) Result {
 	r.Ok = r.Ok && outcome
 	if len(r.Reason) > 0 {
-		r.Reason += "\n\t\t"
+		r.Reason += "\n"
 	}
 	r.Reason += reason
 	return *r
 }
 
+type CheckType string
+
+type Check struct {
+	Type CheckType
+	Func CheckFunc
+}
+
 type CheckFunc func(uri string, config *viper.Viper) (Result, error)
 
 type Registry interface {
-	Get(name string) (CheckFunc, bool)
-	Add(name string, checkFunc CheckFunc) Registry
+	Get(name string) (Check, bool)
+	Add(name string, check Check) Registry
```

Comment:
could `Check` struct have `name` field, so that `Add` can have only single param,  e.g. `Add(check Check)`?

Reply:
Added. This must be a go v Java thing :-), I still think in Java lol